hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestObserverNode.java

```diff
@@ -156,14 +155,17 @@ public void testObserverRequeue() throws Exception {
           assertSentTo(2);
           return fileStatus;
         }, 0, TimeUnit.SECONDS);
-    Thread.sleep(1000);
-    observerEditlogTailer.doTailEdits();
-    FileStatus fileStatus = scheduledFuture.get(1000, TimeUnit.MILLISECONDS);
+
+    GenericTestUtils.waitFor(() -> obRpcMetrics.getRpcRequeueCalls() > oldRequeueNum,
+        50, 10000);
+
+    observerFsNS.getEditLogTailer().doTailEdits();
+    FileStatus fileStatus = scheduledFuture.get(10000, TimeUnit.MILLISECONDS);
     assertNotNull(fileStatus);
 
-    assertTrue(obRpcMetrics.getRpcRequeueCalls() > oldRequeueNum);
-    dfsCluster.getNameNode(2).getNamesystem()
-        .startNewEditLogTailer(dfsCluster.getConfiguration(2));
+    EditLogTailer editLogTailer = new EditLogTailer(observerFsNS, conf);
+    observerFsNS.setEditLogTailerForTests(editLogTailer);
+    editLogTailer.start();
```

Comment:
Just realized we should do this in a try-finally since we are trying to clean up our modifications for other tests, e.g. right now if the `waitFor()` call fails then we won't reset the `editLogTailer` and other tests may fail as a result